Ru language poorly supported

Using 2.0.2 version in Unit test

# **To reproduce error**
val addressProvider = Faker("ru").address

val address = Address(
    territory = addressProvider.city(),
    populatedArea = addressProvider.state(),
    street = addressProvider.streetAddress(),
    building = addressProvider.buildingNumber(),
    block = (1..10).random(),
    flatNumber = (1..100).random()
)

# Error:
```
Unsupported method 'city_name' for component 'address'
java.lang.IllegalArgumentException: Unsupported method 'city_name' for component 'address'
	at io.bloco.faker.FakerComponent.callMethod(FakerComponent.kt:80)
	at io.bloco.faker.FakerComponent.call(FakerComponent.kt:39)
	at io.bloco.faker.FakerComponent$parse$1.invoke(FakerComponent.kt:32)
	at io.bloco.faker.FakerComponent$parse$1.invoke(FakerComponent.kt:30)
	at kotlin.text.Regex.replace(Regex.kt:193)
	at io.bloco.faker.FakerComponent.parse(FakerComponent.kt:30)
	at io.bloco.faker.components.Address.city(Address.kt:52)
	at FakeAssetsManagerTest.generateCall(FakeAssetsManagerTest.kt:69)
	at FakeAssetsManagerTest.generateTherapistCalls(FakeAssetsManagerTest.kt:56)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:108)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:40)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:60)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:52)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy1/jdk.proxy1.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.lang.NoSuchMethodException: io.bloco.faker.components.Address.cityName()
	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2675)
	at io.bloco.faker.FakerComponent.callMethod(FakerComponent.kt:78)
	... 51 more
```

# **Notice**
Not only the address is generated poorly, but also other fields like companies info or first name, which gives the result #{male_first_name}.
When I switch to Dutch lang (nl) it works fine

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -26,7 +26,7 @@ dependencies {
 }
 
 group = "io.bloco"
-version = "2.0.2"
+version = "2.0.3"
 
 java {
     withSourcesJar()

--- a/src/main/kotlin/io/bloco/faker/FakerComponent.kt
+++ b/src/main/kotlin/io/bloco/faker/FakerComponent.kt
@@ -75,7 +75,9 @@ abstract class FakerComponent(private val data: FakerData) {
     private fun callMethod(methodKey: String): String {
         val methodKeyCamel: String = methodKey.snakeToCamel()
         return try {
-            javaClass.getDeclaredMethod(methodKeyCamel).invoke(this) as String
+            javaClass.getDeclaredMethod(methodKeyCamel)
+                .also { it.isAccessible = true }
+                .invoke(this) as String
         } catch (e: NoSuchMethodException) {
             throw IllegalArgumentException("Unsupported method '$methodKey' for component '$key'", e)
         } catch (e: IllegalAccessException) {

--- a/src/main/kotlin/io/bloco/faker/components/Address.kt
+++ b/src/main/kotlin/io/bloco/faker/components/Address.kt
@@ -52,6 +52,10 @@ class Address(data: FakerData) : FakerComponent(data) {
         return parse(fetch("address.city"))
     }
 
+    private fun cityName(): String {
+        return parse("address.city_name")
+    }
+
     fun streetName(): String {
         return parse(fetch("address.street_name"))
     }

--- a/src/main/kotlin/io/bloco/faker/components/Company.kt
+++ b/src/main/kotlin/io/bloco/faker/components/Company.kt
@@ -9,6 +9,10 @@ class Company(data: FakerData) : FakerComponent(data) {
         return parse(fetch("company.name"))
     }
 
+    private fun prefix(): String {
+        return fetch("company.prefix")
+    }
+
     fun suffix(): String {
         return fetch("company.suffix")
     }

--- a/src/main/kotlin/io/bloco/faker/components/Name.kt
+++ b/src/main/kotlin/io/bloco/faker/components/Name.kt
@@ -4,40 +4,31 @@ import io.bloco.faker.FakerComponent
 import io.bloco.faker.FakerData
 
 class Name(data: FakerData) : FakerComponent(data) {
-    fun firstName(): String {
-        return fetch("name.first_name")
-    }
 
-    fun lastName(): String {
-        return fetch("name.last_name")
-    }
+    fun firstName(): String = fetch("name.first_name")
 
-    fun prefix(): String {
-        return fetch("name.prefix")
-    }
+    fun lastName(): String = fetch("name.last_name")
 
-    fun suffix(): String {
-        return fetch("name.suffix")
-    }
+    private fun maleFirstName(): String = fetch("name.male_first_name")
+    private fun femaleFirstName(): String = fetch("name.female_first_name")
+    private fun maleMiddleName(): String = fetch("name.male_middle_name")
+    private fun femaleMiddleName(): String = fetch("name.female_middle_name")
+    private fun maleLastName(): String = fetch("name.male_last_name")
+    private fun femaleLastName(): String = fetch("name.female_last_name")
 
-    fun title(): String {
-        return (
-            fetch("name.title.descriptor") +
-                " " + fetch("name.title.level") +
-                " " + fetch("name.title.job")
-            )
-    }
+    fun prefix(): String = fetch("name.prefix")
 
-    fun name(): String {
-        return parse(fetch("name.name"))
-    }
+    fun suffix(): String = fetch("name.suffix")
 
-    fun nameWithMiddle(): String {
-        return parse(fetch("name.name_with_middle"))
-    }
+    fun title(): String =
+        fetch("name.title.descriptor") +
+            " " + fetch("name.title.level") +
+            " " + fetch("name.title.job")
+
+    fun name(): String = parse(fetch("name.name"))
+
+    fun nameWithMiddle(): String = parse(fetch("name.name_with_middle"))
 
     @Suppress("UNCHECKED_CAST")
-    fun jobTitles(): List<String> {
-        return getMap("name", "title")["job"] as List<String>
-    }
+    fun jobTitles(): List<String> = getMap("name", "title")["job"] as List<String>
 }

--- a/src/test/kotlin/io/bloco/faker/components/AddressTest.kt
+++ b/src/test/kotlin/io/bloco/faker/components/AddressTest.kt
@@ -68,6 +68,16 @@ class AddressTest {
     }
 
     @Test
+    fun city_pt() {
+        assertNotNull(Faker("pt").address.city())
+    }
+
+    @Test
+    fun city_ru() {
+        assertNotNull(Faker("ru").address.city())
+    }
+
+    @Test
     fun streetName() {
         assertNotNull(faker.address.streetName())
     }

--- a/src/test/kotlin/io/bloco/faker/components/CompanyTest.kt
+++ b/src/test/kotlin/io/bloco/faker/components/CompanyTest.kt
@@ -2,6 +2,7 @@ package io.bloco.faker.components
 
 import io.bloco.faker.Faker
 import junit.framework.TestCase
+import junit.framework.TestCase.assertFalse
 import junit.framework.TestCase.assertNotNull
 import org.junit.Test
 
@@ -11,6 +12,13 @@ class CompanyTest {
     @Test
     fun name() {
         assertNotNull(faker.company.name())
+    }
+
+    @Test
+    fun name_ru() {
+        val name = Faker("ru").company.name()
+        assertNotNull(name)
+        assertFalse(name.contains("#"))
     }
 
     @Test

--- a/src/test/kotlin/io/bloco/faker/components/NameTest.kt
+++ b/src/test/kotlin/io/bloco/faker/components/NameTest.kt
@@ -1,6 +1,7 @@
 package io.bloco.faker.components
 
 import io.bloco.faker.Faker
+import junit.framework.TestCase.assertFalse
 import junit.framework.TestCase.assertNotNull
 import junit.framework.TestCase.assertTrue
 import org.junit.Test
@@ -36,6 +37,13 @@ class NameTest {
     @Test
     fun name() {
         assertNotNull(faker.name.name())
+    }
+
+    @Test
+    fun name_ru() {
+        val name = Faker("ru").name.name()
+        assertNotNull(name)
+        assertFalse(name.contains("#"))
     }
 
     @Test

